### DISCUSSION TITLE
Error: Unable to execute dev tunnel operation 'create'. read ECONNRESET

### DISCUSSION THREAD

**Discussion by macik1423** [OfficeDev/microsoft-365-agents-toolkit#10783] [Q&A] [answered]
I have issue 
```
[Error] - code:Ext.DevTunnelOperationError, message: Unable to execute dev tunnel operation 'create'. read ECONNRESET
 Help link: https://aka.ms/teamsfx-local-tunnel-task
```
when I run from "View Samples" -> "Adaptive Card Notification". I know there is a problem with my company network, when I switch to my hot spot everything works. How can I describe problem to my company IT support that dev tunnel cannot run? What is running in the background? Should any proxy create or something else and my company firewall is blocking it, I am wondering what is behind the scenes?

Edit: This is the "Start local tunnel" from `tasks.json`
```
{
    // Start the local tunnel service to forward public URL to local port and inspect traffic.
    // See https://aka.ms/teamsfx-tasks/local-tunnel for the detailed args definitions.
    "label": "Start local tunnel",
    "type": "teamsfx",
    "command": "debug-start-local-tunnel", 
    "args": {
        "type": "dev-tunnel",
        "ports": [
            {
                "portNumber": 3978,
                "protocol": "http",
                "access": "public",
                "writeToEnvironmentFile": {
                    "endpoint": "BOT_ENDPOINT", // output tunnel endpoint as BOT_ENDPOINT
                    "domain": "BOT_DOMAIN" // output tunnel domain as BOT_DOMAIN
                }
            }
        ],
        "env": "local"
    },
    "isBackground": true,
    "problemMatcher": "$teamsfx-local-tunnel-watch"
},
```
Output from terminal:
```
 *  Executing task: Start local tunnel 

Starting local tunnel service

Dev tunnel tag: TeamsToolkitCreatedTag.

Failed to start local tunnel service.
Unable to execute dev tunnel operation 'create'. read ECONNRESET

 *  The terminal process terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.
```

**Comment by macik1423** [comment]
I think I know what's going on, the dev tunnel try to connect with proxy `euw.devtunnels.ms` and my company firewall blocked it.

**Reply by therealjohn** [reply, marked as answer]
This is likely the issue. In the meantime, you could try using the App Test Tool (https://devblogs.microsoft.com/microsoft365dev/introducing-teams-app-test-tool) if it has the features you want to develop in your bot.

**Reply by macik1423** [reply]
I have one more question. If my company's firewall blocked this proxy, will it also block the bot when I publish the app in Teams company microsoft365?

**Reply by therealjohn** [reply]
Unsure, but in a published app you wouldn't be using a dev tunnel. Instead, the bot should point to wherever you are hosting the app, like in Azure. 

You will need to make sure the bot endpoints aren't blocked: https://learn.microsoft.com/en-us/azure/bot-service/bot-service-resources-faq-security?view=azure-bot-service-4.0#which-specific-urls-do-i-need-to-allowlist-in-my-corporate-firewall-to-access-bot-framework-services
